offlinedocs/pages/[[...slug]].tsx

```diff
@@ -194,8 +194,21 @@ const getNavigation = (manifest: Manifest): Nav => {
 	return navigation;
 };
 
-const removeHtmlComments = (string: string) => {
-	return string.replace(/<!--[\s\S]*?-->/g, "");
+const removeHtmlComments = (input: string) => {
+	if (!input) return "";
+
+	let lastResult = input;
+	let currentResult = "";
+
+	// Keep replacing until no more changes occur
+	do {
+		currentResult = lastResult.replace(/<!--[\s\S]*?-->/g, "");
+		if (currentResult === lastResult) {
+			break;
+		}
+		lastResult = currentResult;
+	} while (true);
+	return currentResult;
```

Comment:
```suggestion
const removeHtmlComments = (input: string) => {
	return string.replaceAll(/<!--[\s\S]*?-->/g, "");
```

I think this is what you're really after :p

Reply:
The problem is that regex doesn't handle nested comments which is what CodeQL is complaining about. The motivation behind the function is to iterate over the string until all comments have been stripped regardless of how nested they are.

Reply:
the code here, to my eye at least, doesn't do anything to handle "nesting". it just calls `.replace` over and over again until it stops changing the string, which is exactly what `.replaceAll` does

Reply:
Ahh I missed the `All`. I think that will probably work although I noticed the tool recommended using `html-sanitize` which I think will also solve this and is less likely to trip the tool up in the future.